crypto/sm2/sm2_aid.c

```diff
@@ -0,0 +1,45 @@
+/*
```

Comment:
This file is not necessary, is it?

Reply:
Why this is not needed?

Reply:
Because you're not using `sm2_algorithmidentifier_encoding()` anywhere, and you have `DER_w_algorithmIdentifier_SM2_with_MD()` (in `providers/common/der/der_sm2_sig.c`), which you *do* use, and which does the same thing, but generated from actual ASN.1 code so you don't have to re-create DER code macros all over the place.

Reply:
While looking into this, I just noticed we still declare similar functions for other algos.  I'll clean those away...